handler/src/main/java/io/netty/handler/ipfilter/IpSubnetFilter.java

```diff
@@ -0,0 +1,150 @@
+        this(true, Arrays.asList(ObjectUtil.checkNotNull(rules, "rules")));
+    }
+
+    /**
+     * <p> Create new {@link IpSubnetFilter} Instance with specified {@link IpSubnetFilterRule} as array
+     * and specify if we'll accept a connection if we don't find it in the rule(s). </p>
+     *
+     * @param acceptIfNotFound {@code true} if we'll accept connection if not found in rule(s).
+     * @param rules            {@link IpSubnetFilterRule} as array
+     */
+    public IpSubnetFilter(boolean acceptIfNotFound, IpSubnetFilterRule... rules) {
+        this(acceptIfNotFound, Arrays.asList(ObjectUtil.checkNotNull(rules, "rules")));
+    }
+
+    /**
+     * <p> Create new {@link IpSubnetFilter} Instance with specified {@link IpSubnetFilterRule} as {@link List}. </p>
+     * <p> {@code acceptIfNotFound} is set to {@code true} </p>
+     *
+     * @param rules {@link IpSubnetFilterRule} as {@link List}
+     */
+    public IpSubnetFilter(List<IpSubnetFilterRule> rules) {
+        this(true, rules);
+    }
+
+    /**
+     * <p> Create new {@link IpSubnetFilter} Instance with specified {@link IpSubnetFilterRule} as {@link List}
+     * and specify if we'll accept a connection if we don't find it in the rule(s). </p>
+     *
+     * @param acceptIfNotFound {@code true} if we'll accept connection if not found in rule(s).
+     * @param rules            {@link IpSubnetFilterRule} as {@link List}
+     */
+    public IpSubnetFilter(boolean acceptIfNotFound, List<IpSubnetFilterRule> rules) {
+        this.rules = ObjectUtil.checkNotNull(rules, "rules");
+        this.acceptIfNotFound = acceptIfNotFound;
+
+        // Iterate over rules and check for `null` rule.
+        for (IpSubnetFilterRule ipSubnetFilterRule : this.rules) {
+            ObjectUtil.checkNotNull(ipSubnetFilterRule, "rule");
+        }
+
+        sortAndFilter();
+    }
+
+    @Override
+    protected boolean accept(ChannelHandlerContext ctx, InetSocketAddress remoteAddress) {
+        int indexOf = Collections.binarySearch(this.rules, remoteAddress, IpSubnetFilterRuleComparator.INSTANCE);
+        if (indexOf >= 0) {
+            return this.rules.get(indexOf).ruleType() == IpFilterRuleType.ACCEPT;
+        }
+        return acceptIfNotFound;
+    }
+
+    /**
+     * <ol>
+     *     <li> Sort the list </li>
+     *     <li> Remove over-lapping subnet </li>
+     *     <li> Sort the list again </li>
+     * </ol>
+     */
+    private void sortAndFilter() {
+        Collections.sort(this.rules);
+        Iterator<IpSubnetFilterRule> iterator = rules.iterator();
+        List<IpSubnetFilterRule> toRemove = new ArrayList<IpSubnetFilterRule>();
+
+        try {
+            IpSubnetFilterRule parentRule = null;
+            while (iterator.hasNext()) {
+
+                // If parentRule is null, take first element out of Iterator.
+                if (parentRule == null) {
+                    parentRule = iterator.next();
+                }
```

Comment:
Please do another `hasNext` check, instead of relying on catching `NoSuchElementException`.

Reply:
`hasNext` is already done in `while(Iterator#hasNext())` so no need to do the same here but I'll do it in other ones.